flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReaderTest.java

```diff
@@ -115,6 +128,44 @@ public void testWakeUp() throws Exception {
         assertNull(error.get());
     }
 
+    @Test
+    public void testNumBytesInCounter() throws Exception {
+        // Create a task metric group for intercepting numBytesOut
+        final InterceptingTaskMetricGroup taskMetricGroup = new InterceptingTaskMetricGroup();
+        final OperatorMetricGroup operatorMetricGroup =
+                new InternalOperatorMetricGroup(
```

Comment:
Can you rebase and check if this constructor is still public?

Reply:
I have done https://github.com/fapaul/flink/commit/01de3e6db4c68d316d02bda35279e163673aa2bc to circumvent the problem I think you can do similarly for the source metric group.

Reply:
Thanks @fapaul for the reminder! I made another detour in 744fcd3